dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/DefaultProgramRuleVariableService.java

```diff
@@ -65,6 +71,7 @@ public DefaultProgramRuleVariableService( ProgramRuleVariableStore programRuleVa
     public long addProgramRuleVariable( ProgramRuleVariable programRuleVariable )
     {
         programRuleVariableStore.save( programRuleVariable );
+        programRuleVariablesCache.invalidateAll();
```

Comment:
This is okay, though I don't think this makes it solid as we use the metadata importer to import all metadata including ProgramRuleVariables, which means the cache won't be cleared mostly. This then gives a "false sense of security". 

Could we maybe instead just introduce a `isLinkedToProgramRuleVariableCached` method where it is explicit that the caller is getting a cached result for now? Or just remove the invalidation call and add the caching behavior in the Java docs of this method?

Reply:
The method is used only in the postprocessor, so I just renamed the method.
And I maintained the invalidation in the delete method that is called in the deletion handlers